client/roomlistdock.cpp

```diff
@@ -194,10 +195,14 @@ RoomListDock::RoomListDock(MainWindow* parent)
     forgetAction =
         roomContextMenu->addAction(QIcon::fromTheme("irc-remove-operator"),
         tr("Forget room"), this, [this] {
-            if (auto room = getSelectedRoom())
-            {
-                Q_ASSERT(room->connection());
-                room->connection()->forgetRoom(room->id());
+            if (auto room = getSelectedRoom()) {
+                QMessageBox::StandardButton confirmation = QMessageBox::question(
+                    this, tr("Forget this room?"),
+                    tr("Are you sure you want to forget room %1?").arg(room->name()));
+                if (confirmation == QMessageBox::Yes) {
+                    Q_ASSERT(room->connection());
+                    room->connection()->forgetRoom(room->id());
```

Comment:
```suggestion
                    if (QUO_CHECK(room->connection())
                        room->connection()->forgetRoom(room->id());
```
We have a safer alternative to `Q_ASSERT` in libQuotient.